test/Microsoft.Health.Fhir.R4.Tests.E2E/GraphQl/GraphQlTests.cs

```diff
@@ -0,0 +1,183 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Rest;
+using Microsoft.Health.Fhir.Tests.E2E.Rest.Search;
+using Microsoft.Health.Test.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.R4.Tests.E2E.GraphQl
+{
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.Json)]
+    public class GraphQlTests : SearchTestsBase<HttpIntegrationTestFixture>
+    {
+        public GraphQlTests(HttpIntegrationTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        [Fact]
+        public async void GivenQueryByGet_AskingForGraphQlSchema_ThenGraphQlSchemaShouldBeReturned()
+        {
+            using HttpResponseMessage response = await Client.HttpClient.GetAsync("graphql/graphql?sdl");
+            response.EnsureSuccessStatusCode();
+
+            string schema = response.Content.ReadAsStringAsync().Result;
+            Assert.Contains("Patient", schema);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async void GivenPatients_WhenFetchingByGraphQl_ThenAllPatientsInTheServerShouldBeReturned()
+        {
+            // Create various resources.
+            var tag = Guid.NewGuid().ToString();
+            Patient[] patients = await Client.CreateResourcesAsync<Patient>(
+                p => SetPatientInfo(p, "Seattle", "Robinson", tag),
+                p => SetPatientInfo(p, "Portland", "Williamas", tag),
+                p => SetPatientInfo(p, "Seattle", "Jones", tag));
+
+            // Define query.
+            var queryObject = new
+            {
+                query = @"query { 
+                            patients { 
+                                id
+                                name {
+                                    family
+                                }
+                            }
+                        }",
+                variables = new { },
+            };
+            string jsonString = JsonConvert.SerializeObject(queryObject);
+            var httpContent = new StringContent(jsonString, Encoding.UTF8, "application/json");
+
+            HttpResponseMessage response = await Client.HttpClient.PostAsync("graphql", httpContent);
+            response.EnsureSuccessStatusCode();
+
+            // Ensure 3 patients are returned.
+            var responseString = await response.Content.ReadAsStringAsync();
+            dynamic responseObj = JsonConvert.DeserializeObject<dynamic>(responseString);
+
+            var patientsResponse = ((JArray)responseObj.data.patients).Select(x => new Patient
+            {
+                Id = (string)x["id"],
+                Name = x["name"].ToObject<List<HumanName>>(),
+            }).ToList();
+
+            Assert.NotNull(patientsResponse);
```

Comment:
Could you assert that the length of the list returned is equal to or greater than 3?

Reply:
Done! :)